.github/workflows/test.yml

```diff
@@ -6,6 +6,10 @@ on:
     branches:
       - master
 
+concurrency:
+  group: ${{ github.ref }}
```

Comment:
Also this option needs to include the workflow name, otherwise it cancels all in-flight workflows regardless of name.

> If you have multiple workflows in the same repository, concurrency group names must be unique across workflows to avoid canceling in-progress jobs or runs from other workflows. Otherwise, any previously in-progress or pending job will be canceled, regardless of the workflow.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency

I'll see about adjusting this.

Reply:
https://github.com/vectordotdev/vector/pull/12740